blockchain/state_manager/src/lib.rs

```diff
@@ -223,10 +262,238 @@ where
                 .collect::<Result<Vec<Block>, _>>()?;
             // convert tipset to fulltipset
             let full_tipset = FullTipset::new(blocks)?;
-            self.apply_blocks(&full_tipset, &chain_rand)
+            self.apply_blocks(&full_tipset, &chain_rand, call_back)
         })
     }
 
+    pub fn look_up_id(&self, addr: &Address, tipset: &Tipset) -> Result<Address, Error> {
+        let tipset = ChainStore::new(self.get_block_store())
```

Comment:
Dont need to do all this. If you look at the tipset_from_keys method, its a wrapper for a function that does not take in the chainstore but just the blockstore